### ISSUE TITLE
ClassMap::getAmbiguousClasses not working at all

### ISSUE THREAD

**Issue by janedbal** [composer/class-map-generator#3]
Following script dumps empty array:

```php
$generator = new ClassMapGenerator(['php']);
$generator->scanPaths(__DIR__ . '/sample-collisions');

var_dump($generator->getClassMap()->getAmbiguousClasses());
```

while folder `sample-collissions` contains only `file1.php` and `file2.php` both with equal contents:

```php
<?php
class GlobalClass1 {}
```

**Comment by Seldaek** [comment]
Can you create a demo git repo that reproduces the problem? Because as far as I can tell this definitely should work..

**Comment by janedbal** [comment]
It works, not sure what my previous error was. Sorry for confusion.

---------

Anyway, the method `getAmbiguousClasses` is documented with `A map of class names to their list of ambiguous paths`, so I'd expect the output to be:
```
["GlobalClass1" => ["sample-collisions/file1.php", "sample-collisions/file2.php"]]
```
but it is
```
["GlobalClass1" => ["sample-collisions/file2.php"]]
```

Your test even [asserts that](https://github.com/composer/class-map-generator/blob/main/tests/ClassMapGeneratorTest.php#L173). Is that really intentional? That way I'm not really able to detect where the problem is.

**Comment by Seldaek** [comment]
Yes you need to get the class's resolved path in the class map to get what files the conflicting files conflicted with. This allows you to know which was selected and which was/were ignored.

**Comment by janedbal** [comment]
I see, thx, closing
